src/Castle.Core/Components.DictionaryAdapter/DictionaryAdapterFactory.cs

```diff
@@ -171,7 +175,8 @@ private object InternalGetAdapter(Type type, IDictionary dictionary, PropertyDes
 		}
 
 		#region Type Builders
-	
+
+#if FEATURE_APPDOMAIN

```

Comment:
Since .NET 4.5 has `AssemblyBuilder.DefineDynamicAssembly` and `AssemblyBuilder.DefineDynamicModule` this should be `FEATURE_LEGACY_REFLECTION_API` instead so we can easily drop the old code once we no longer support pre-.NET4.5.